tools/autograd/derivatives.yaml

```diff
@@ -277,7 +277,7 @@
   output_differentiability: [False]
 
 - name: acosh(Tensor self) -> Tensor
-  self: grad * (self.pow(2) - 1).rsqrt().conj()
+  self: "self.is_complex() ? grad * ((self + 1).rsqrt() * (self - 1).rsqrt()).conj() : grad * (self * self - 1).rsqrt()"
```

Comment:
The two are mathematically the same at a glance? Why can't we keep only one?

Reply:
The first one is correct for both real and complex numbers. The second one is a simplification for real numbers.

Reply:
Can you leave a comment there: save one addition by re-ordering the terms for the real case.

Reply:
Ugh, of course I wrote "compelx"... lol

Reply:
This can be added to any other PR :D